website_sale_google_shopping/controllers/website_sale_google_shopping.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# License, author and contributors information in:
+# __openerp__.py file at the root folder of this module.
+from datetime import datetime, timedelta
+import openerp
+from functools import partial
+from openerp import http
+from openerp.http import request
+from openerp.addons.website_sale.controllers.main import get_pricelist
+
+
+DATE_FORMAT = '%Y-%m-%d %H:%M'
+FEED_MIMETYPE = 'application/xml; charset=utf-8'
+
+
+class GoogleShopping(http.Controller):
+    def get_effective_date(self, date_start, date_end):
+        if not date_start or not date_end:
+            return ''
+        ds = datetime.strptime(date_start, '%Y-%m-%d %H:%M:%S')
+        de = datetime.strptime(date_end, '%Y-%m-%d %H:%M:%S')
+        return '%sT%s/%sT%s' % (
+            ds.strftime('%Y-%m-%d'),
+            ds.strftime('%H:%M'),
+            de.strftime('%Y-%m-%d'),
+            de.strftime('%H:%M'))
+
+    def generate_feed(self):
+        env = request.env
+        context = request.context
+        url = '/google-shopping.xml'
+        content = None
+        ir_attachment = env['ir.attachment']
+        attachments = ir_attachment.sudo().search_read([
```

Comment:
I feel like this sudo could have some unintended side-effects. Shouldn't the public user be able to access all attachments related to this feed?

Also, any particular reason for `search_read` instead of just `search`? You would then be able to utilize it like a recordset instead of the list/dictionary combo below - much more Pythonic IMO